File: backend/domain/src/commonMain/kotlin/me/nathanfallet/shortt/domain/usecases/links/CreateLinkUseCaseImpl.kt
package me.nathanfallet.shortt.domain.usecases.links

import me.nathanfallet.shortt.domain.models.links.Link
import me.nathanfallet.shortt.domain.repositories.links.LinksRepository
import kotlin.uuid.Uuid

class CreateLinkUseCaseImpl(
    private val linksRepository: LinksRepository,
) : CreateLinkUseCase {
    override suspend fun invoke(url: String, slug: String?, userId: Uuid): Link {
        val finalSlug = slug ?: ""
        // TODO: Validate slug uniqueness + slug generation logic
        return linksRepository.create(url, finalSlug, userId)
    }
}

File: frontend/webApp/src/domain/usecases/links/CreateLinkUseCaseImpl.ts
import {CreateLinkUseCase} from "./CreateLinkUseCase.ts";
import {Link} from "../../models/Link.ts";
import {LinksRepository} from "../../repositories/LinksRepository.ts";

export class CreateLinkUseCaseImpl implements CreateLinkUseCase {
    constructor(private repository: LinksRepository) {
    }

    async invoke(url: string, slug: string | null): Promise<Link> {
        return this.repository.create(url, slug);
    }
}
